### ISSUE TITLE
Crashes when both suite and overrideTestDescription settings are used

### ISSUE THREAD

**Issue by Agnaro** [tornaia/karma-sonarqube-unit-reporter#49]
When both the `suite` and `overrideTestDescription` options are set, the following line throws an error.
```javascript
lastFilePath = fileNodes.children[numberOfFileNodes - 1].attributes.getNamedItem('path').value
```
Error:
```
TypeError: Cannot read property 'value' of undefined
    at SonarQubeUnitReporter.specSuccess.specSkipped.specFailure (.../node_modules/karma-sonarqube-unit-reporter/index.js:157:95)
    at SonarQubeUnitReporter.BaseReporter.onSpecComplete (.../node_modules/karma/lib/reporters/base.js:107:12)
    at Server.<anonymous> (.../node_modules/karma/lib/events.js:40:26)
```

For some reason, the <file> tag does not have a 'path' attribute in this case.
```javascript
XMLElement {
  parent: XMLElement {...},
  options: {...},
  stringify: XMLStringifier {...},
  value: null,
  children: [
    XMLElement {
      parent: [Circular],
      options: [Object],
      stringify: [XMLStringifier],
      value: null,
      children: [Array],
      baseURI: null,
      name: 'testCase',
      type: 1,
      attribs: [Object],
      schemaTypeInfo: null
    }
  ],
  baseURI: null,
  name: 'file',
  type: 1,
  attribs: {},
  schemaTypeInfo: null,
  attributeMap: XMLNamedNodeMap { nodes: {} }
}

```

**Comment by DincaAdrian1991** [comment]
Hi @Agnaro , I've had the same issue and after hours of bashing my head in, I found what was the problem. I've had one of the test files named incorrectly. So my problem was something like "controller-spec.js", while the pluggin tries to match the test results with the files. While not giving another `testFilePattern`(which has default: "/(\.spec\.ts|\.spec.js)/") as described here
https://github.com/tornaia/karma-sonarqube-unit-reporter/blob/83cd9a8c09e174427b26a793e48b609bcbe90088/index.js#L216

It means that it will have test results on files it cannot detect and that will generate this issue. 
My solution was to rename the file correctly, but you can also overwrite that `testFilePattern` to something like spec.js|spec.ts without the dot.

Can you please try this out and confirm or close the issue?

**Comment by Agnaro** [comment]
@DincaAdrian1991 I'm still getting the same error if i set `testFilePattern=".ts` to include all typescript files (we only use ts files).  I don't think we are running tests in files that don't match the default file pattern since we are using Angular and `ng test` and that is set up to only run files that match the regex `/\.spec\.ts$/`.  I can't be 100% sure though since we do have a lot of spec files.

**Comment by DincaAdrian1991** [comment]
So what I did in my case, I've replaced this part where it was crashing from `index.js`
```      lastFilePath = fileNodes.children[numberOfFileNodes - 1].attributes.getNamedItem('path').value
      if (lastFilePath !== nextPath) {
        suites[browser.id].ele('file', {
          path: nextPath,
        })
      }
```

with this part:

```
      try {
        lastFilePath = fileNodes.children[numberOfFileNodes - 1].attributes.getNamedItem('path').value;
        if (lastFilePath !== nextPath) {
          suites[browser.id].ele('file', {
            path: nextPath,
          })
        }
      } catch (error) {
        suites[browser.id].ele('file', {
          path: 'FILE_ABOVE_DOES_NOT_HAVE_A_PATH',
        })
      }
```

Then go to the resulted xml report and search for `FILE_ABOVE_DOES_NOT_HAVE_A_PATH`, the file above will have no path and you can identify it by it's test cases. Once you identified the file/problem, you can revert the changes.

 Hope this helps you!

**Comment by Agnaro** [comment]
@DincaAdrian1991 That was a good suggestion, but it turns out that it's not just a single file that it can't find.  It is every file.  This is because I'm using a yarn workspace (monorepo) and running the tests inside one of the packages.  My goal was to have the paths in the XML relative to the workspace root rather than the package root (which is what SonarQube needs).  I think i misunderstood what the `suite` option did.  I thought `suite` got prepended to the path, but it is actually being prepended to the "describe path" of the test.

So all of this is because I wasn't using `suite` right.  I do have a workaround that uses `filenameFormatter`.  Thank you for your help.

**Comment by tornaia** [comment]
Fixed in 1.0.0 (#75). The reporter no longer reads the XML back while collecting results, so this crash cannot happen any more; see [CHANGELOG.md](https://github.com/tornaia/karma-sonarqube-unit-reporter/blob/master/CHANGELOG.md) for the full list of changes. Please reopen if you still hit it with 1.0.0.
